src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java

```diff
@@ -922,8 +931,17 @@ private void addRpcStubImports(InterfaceContext context) {
         typeTable.saveNicknameFor("com.google.api.client.http.HttpMethods");
         typeTable.saveNicknameFor("com.google.api.core.InternalApi");
         typeTable.saveNicknameFor("com.google.api.pathtemplate.PathTemplate");
-        typeTable.saveNicknameFor("com.google.api.gax.httpjson.ApiMessageHttpRequestFormatter");
-        typeTable.saveNicknameFor("com.google.api.gax.httpjson.ApiMessageHttpResponseParser");
+        String configSchemaVersion = context.getProductConfig().getConfigSchemaVersion();
+        if (configSchemaVersion != null && configSchemaVersion.startsWith("1.")) {
```

Comment:
Is this also to distinguish between DiscoGAPIC and REGAPIC? If so, add a comment to make that explicit.

Reply:
This is correct. A hacky way of doing it, but it is quite reliable way to distinguish (DIREGAPIC is always 2.0 gapic yaml or no yaml, while discogapic is always gapic yaml 1.0).

Reply:
Yeah. The comment is very helpful! (As in the other file where you also added it.) Thanks!